src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -182,32 +182,45 @@ def _alloc_node(self, node, requested_cores, requested_gpus,
         requested (but the call will never return more than requested).
         '''
 
-
+        # Create empty list of cores and gpus. This empty list will be populated
+        # with core ids and gpu ids if they are available in the current node.
         cores = list()
         gpus  = list()
 
+        # Count the number of free cores and gpus in the current node. This is 
+        # way quicker than actually finding the core IDs.
         free_cores = node['cores'].count(rpc.FREE)
         free_gpus  = node['gpus' ].count(rpc.FREE)
 
-        # first count the free cores/gpus, as that is way quicker than
-        # actually finding the core IDs.
+
         if partial:
-            # For partial requests the check simpliefies: we just check if we 
-            # have either, some cores *or* gpus, to serve the request
+            # For partial requests, the check simplifies: we need to check if we 
+            # have either some cores *or* gpus to serve the request, if yes we
+            # continue. 
+            # In the following line, we check for its equivalent negative logic:
+            # If both no. of cpus and no. of gpus free on this node do not satisfy the 
+            # request (i.e. there are no free cores and no free gpus), return.
+            # NOTE: This condition needs to checked only for MPI units.
             if  (requested_cores and not free_cores) and \
-                (requested_gpus  and not free_gpus )     :
-                # wa can't serve either request
+                (requested_gpus  and not free_gpus ):
                 return [], []
 
         else:
-            # non-partial requests (ie. full requests): check if we can serve
-            # both, requested cores *and* gpus.
+            # For non-partial requests (ie. full requests): we need to check if 
+            # we can serve both, requested cores *and* gpus, if yes we continue.
+            # In the following line, we check for its equivalent negative logic:
+            # If both no. of cpus and no. of gpus free on this node do not satisfy the 
+            # request (i.e. there are no. of requested cores is more than the 
+            # no. of free cores and requested gpus if more than the no. of free
+            # gpus), return. 
             if  requested_cores > free_cores or \
                 requested_gpus  > free_gpus     :
                 return [], []
 
```

Comment:
I am not disagreeing with the comment, but is that really a useful level of detail?  We have seven lines of comment for, essentially:
```
  if request_a > resource_a or \
     request_b > resource_b:
        return [],[]
```

Reply:
I agree it is a bit redundant. The idea was to (i) keep the comment symmetrical with the partial case and (ii) explain that, in order to check if we have adequate resources, we are checking for the negative logic in code (or) whereas in the comment we mention the positive logic (and), but both are equivalent. !(A and B) = !A or !B

Reply:
I would have the same general comment on the partial case :)    Agree on the neg logic - but then we could change the comment to 
```
In the non-partial case its a no-match if either the cpu or gpu request cannot be served.
```

Reply:
Ok, I'll change the comment then :)